Lesson_Materials/Matrix_Transpose/index.html

```diff
@@ -0,0 +1,455 @@
+						#### Coalesced global memory
+					</div>
+					<div class="container" data-markdown>
+						![SYCL](./coalesced_global_memory_4.png "SYCL")
+					</div>
+				</section>
+				<!--Slide 8-->
+				<section>
+					<div class="hbox" data-markdown>
+						#### Coalesced global memory
+					</div>
+					<div class="container" data-markdown>
+						![SYCL](./coalesced_global_memory_5.png "SYCL")
+					</div>
+				</section>
+				<!--Slide 9-->
+				<section>
+					<div class="hbox" data-markdown>
+						#### Coalesced global memory
+					</div>
+					<div class="container" data-markdown>
+						![SYCL](./coalesced_global_memory_6.png "SYCL")
+					</div>
+				</section>
+				<!--Slide 10-->
+				<section>
+					<div class="hbox" data-markdown>
+						#### Row-major vs Column-major
+					</div>
+					<div class="container" data-markdown>
+						* Coalescing global memory access is particularly important when working in multiple dimensions.
+						* This is because when doing so you have to convert from a position in 2d space to a linear memory space.
+						* There are two ways to do this; generally referred to as row-major and column-major.
+					</div>
+				</section>
+				<!--Slide 11-->
+				<section>
+					<div class="hbox" data-markdown>
+						#### Row-major vs Column-major
+					</div>
+				</section>
+				<section>
+					<div class="container" data-markdown>
+						![SYCL](./row_col_1.png "SYCL")
+					</div>
+				</section>
+				<!--Slide 12-->
+				<section>
+					<div class="container" data-markdown>
+						![SYCL](./row_col_2.png "SYCL")
+					</div>
+				</section>
+				<!--Slide 13-->
+				<section>
+					<div class="container" data-markdown>
+						![SYCL](./row_col_3.png "SYCL")
+					</div>
+				</section>
+				<!--Slide 14-->
+				<section>
+					<div class="hbox" data-markdown>
+						#### Cost of accessing global memory
+					</div>
+					<div class="container" data-markdown>
+						* Global memory is very expensive to access.
+						* Even with coalesced global memory access if you are accessing the same elements multiple times that can be expensive.
+						* Instead you want to cache those values in a lower latency memory.
+					</div>
+				</section>
+				<!--Slide 15-->
+				<section>
+					<div class="hbox" data-markdown>
+						#### Using local memory
+					</div>
+					<div class="container" data-markdown>
+						![SYCL](./local_memory.png "SYCL")
+					</div>
+					<div class="container" data-markdown>
+						* The solution is local memory.
+						* Local memory is generally on-chip and doesn't have a cache as it's managed manually so is much lower latency.
+						* Local memory is a smaller dedicated region of memory per work-group.
+						* Local memory can be used to cache, allowing us to read from global memory just once and then read from local memory instead, often referred to as a scratchpad.
+						* Local memory can be accessed in an uncoalesced fashion without much performance degradation.
```

Comment:
Do we want to mention bank conflicts here or is that too much for this tutorial?

Reply:
I think that's a bit too advanced. This exercise is meant to be the bare minimum intro to local mem and coalesced global memory